### PR TITLE
Fixing up bundle and make target

### PR DESCRIPTION
**Pull request by ewolinetz**
Updating our make target to be consistent and fix our bundle image generation

### CODE REVIEW

**Comment by ewolinetz** [comment]
currently seeing this in the catalog:
```
time="2020-08-05T20:06:57Z" level=warning msg="couldn't find service in cache" service=elasticsearch-catalog
time="2020-08-05T20:06:57Z" level=info msg="state.Key.Namespace=openshift-operators-redhat state.Key.Name=elasticsearch-catalog state.State=CONNECTING"
time="2020-08-05T20:06:57Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:06:58Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:06:59Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:07:00Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:07:00Z" level=info msg="state.Key.Namespace=openshift-operators-redhat state.Key.Name=elasticsearch-catalog state.State=TRANSIENT_FAILURE"
E0805 20:07:00.981483       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.30.87.165:50051: connect: no route to host"
time="2020-08-05T20:07:01Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
E0805 20:07:01.303513       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.30.87.165:50051: connect: no route to host"
time="2020-08-05T20:07:01Z" level=info msg="state.Key.Namespace=openshift-operators-redhat state.Key.Name=elasticsearch-catalog state.State=CONNECTING"
time="2020-08-05T20:07:02Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:07:02Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:07:17Z" level=info msg="state.Key.Namespace=openshift-operators-redhat state.Key.Name=elasticsearch-catalog state.State=READY"
E0805 20:07:17.313384       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:07:17.324762       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:07:17.345061       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:07:17.435104       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:07:17.607114       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:07:17.937308       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
time="2020-08-05T20:09:50Z" level=info msg="Adding related objects for operator-lifecycle-manager-catalog"
E0805 20:13:18.427139       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:25.429534       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:25.827260       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:26.231114       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:26.626537       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:27.029527       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:27.426945       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:13:27.829015       1 queueinformer_operator.go:290] sync {"update" "openshift-operators-redhat"} failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
time="2020-08-05T20:14:16Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
E0805 20:14:16.840391       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
time="2020-08-05T20:14:17Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
time="2020-08-05T20:14:17Z" level=info msg=syncing event=update reconciling="*v1alpha1.Subscription" selflink=/apis/operators.coreos.com/v1alpha1/namespaces/openshift-operators-redhat/subscriptions/elasticsearch-operator
E0805 20:14:18.631503       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:19.632918       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:20.430512       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:21.028505       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:21.432240       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:21.829106       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:22.231117       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:23.031267       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:23.428611       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:23.831054       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:24.229800       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:24.631963       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:25.028661       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:25.432105       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
E0805 20:14:25.829866       1 queueinformer_operator.go:290] sync "openshift-operators-redhat" failed: {elasticsearch-operator 4.6  {elasticsearch-catalog openshift-operators-redhat}} not found: rpc error: code = Unknown desc = no entry found for elasticsearch-operator 4.6
time="2020-08-05T20:14:55Z" level=info msg="Adding related objects for operator-lifecycle-manager-catalog"
```

**Comment by ewolinetz** [comment]
above logs were resolved by building index image using a publicly available bundle and pushing the index to a publicly available repo

**Comment by ewolinetz** [comment]
will need to open a CLO pr as well afterwards

File: Makefile [hunk 1]
```diff
@@ -119,18 +119,21 @@ uninstall:
...
 		[ ! -d ./operator-registry ] && git clone https://github.com/operator-framework/operator-registry ; \
 		cd operator-registry ; \
-		go build ./cmd/opm/ ; \
+		make build ; \
 		popd ; \
 	fi
 .PHONY: opm
 
+MANIFEST_VERSION?="4.6"
 generate-bundle: opm
```
Comment (periklis) [inline Makefile:133]: I suggest to make `regenerate` a target dependency to ensure that the bundles picks always the latest CRDs.

**Comment by ewolinetz** [comment]
moving updating olm installation work to https://github.com/openshift/elasticsearch-operator/pull/451

**Comment by ewolinetz** [comment]
@jcantrill we'll want a CLO version of this as well. i can pick it up on Monday

**Pull request merged**